frontend/src/libs/packages/middleware/geolocation.middleware.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  type AnyAction,
+  type Dispatch,
+  type ThunkDispatch,
+  type ThunkMiddleware,
+} from '@reduxjs/toolkit';
+
+import { shiftEnded, startShift } from '~/slices/driver/actions.js';
+import { ShiftStatus } from '~/slices/driver/libs/enums/shift-status.enum.js';
+import {
+  restartWatchTruckLocation,
+  startWatchTruckLocation,
+} from '~/slices/trucks/actions.js';
+import { type StartWatchTruckLocation } from '~/slices/trucks/types/start-watch-truck-location.type.js';
+
+import {
+  type ExtraArguments,
+  type RootReducer,
+} from '../store/libs/types/store.types.js';
+import { updateGeolocation } from './libs/helpers/update-geolocation.helper.js';
+
+const geolocationMiddleware: ThunkMiddleware<
+  RootReducer,
+  AnyAction,
+  ExtraArguments
+> = ({
+  dispatch,
+  getState,
+}: {
+  dispatch: ThunkDispatch<RootReducer, ExtraArguments, AnyAction>;
+  getState: () => RootReducer;
+}) => {
+  const locationUpdateIntervals: NodeJS.Timer[] = [];
+
+  const setLocationUpdateInterval = (interval: NodeJS.Timer): void => {
+    for (const oldInterval of locationUpdateIntervals) {
+      clearInterval(oldInterval);
+    }
+    locationUpdateIntervals.push(interval);
+  };
```

Comment:
Why are we using array here? Maybe it'd better without it?

```suggestion
  const locationUpdateInterval: NodeJS.Timer|null = null;

  const setLocationUpdateInterval = (interval: NodeJS.Timer): void => {
      locationUpdateInterval = interval;
  };
```